### PR TITLE
Implement StreamingInterfaceINTEL execution mode

### PR DESCRIPTION
**Pull request by MrSidims**
In an FPGA design, one may want to control the interface that a kernel
exposes to the rest of the design. This extension adds a new Execution
Mode to indicate that the kernel has a streaming interface, in which
invocation and return is synchronized by a handshaking protocol.

The decision whether to add StreamingInterfaceINTEL execution mode with
0/1 literal is based on the kernel metadata, in case of:
!ip_interface !N
!N = !{!"streaming"}
the translator emits StreamingInterfaceINTEL 0

and

!ip_interface !N
!N = !{!"streaming", !"stall_free_return"}
the translator emits StreamingInterfaceINTEL 1

Spec update:
https://github.com/KhronosGroup/SPIRV-Registry/pull/130

Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

**Comment by MrSidims** [comment]
@mlychkov @AlexeySotkin @pauzinl @KornevNikita please take a look

File: lib/SPIRV/PreprocessMetadata.cpp [hunk 1]
```diff
@@ -256,6 +256,35 @@ void PreprocessMetadataBase::visit(Module *M) {
...
+      // !ip_interface !N
+      // !N = !{!"streaming"}
+      // 'stall free' mode (to be mapped on '1' literal)
+      // !ip_interface !N
+      // !N = !{!"streaming", !"stall_free_return"}
+      int32_t InterfaceMode = -1;
+      for (size_t I = 0; I != Interface->getNumOperands(); ++I)
+        InterfaceStrSet.insert(getMDOperandAsString(Interface, I));
+      if (InterfaceStrSet.find("stall_free_return") != InterfaceStrSet.end())
+        InterfaceMode = 1;
```
Comment (mlychkov) [inline lib/SPIRV/PreprocessMetadata.cpp:278]: Should we check here that "streaming" is also present in MD?

Comment (MrSidims) [reply lib/SPIRV/PreprocessMetadata.cpp:278]: Hm, actually, lets check it, thanks!

**Comment by KornevNikita** [comment]
👍🏻

**Review by mlychkov — APPROVED** [review]

**Pull request merged**